packages/subgraph/src/plugin/pluginSetupProcessor.ts

```diff
@@ -7,129 +7,258 @@ import {
   UpdateApplied,
   UpdatePrepared
 } from '../../generated/PluginSetupProcessor/PluginSetupProcessor';
-import {Plugin, PluginHelper} from '../../generated/schema';
-import {addPlugin} from './utils';
+import {
+  PluginInstallation,
+  PluginPermission,
+  PluginPreparation
+} from '../../generated/schema';
+import {
+  addPlugin,
+  getPluginInstallationId,
+  PERMISSION_OPERATIONS
+} from './utils';
 
 export function handleInstallationPrepared(event: InstallationPrepared): void {
-  let pluginId = event.params.plugin.toHexString();
-
-  let pluginEntity = new Plugin(pluginId);
-  pluginEntity.sender = event.params.sender.toHexString();
-  pluginEntity.dao = event.params.dao.toHexString();
-  // TODO: SARKAWT
-  // pluginEntity.pluginSetup = event.params.pluginSetup.toHexString();
-  pluginEntity.data = event.params.data;
-  pluginEntity.state = 'InstallationPrepared';
+  let dao = event.params.dao.toHexString();
+  let plugin = event.params.plugin.toHexString();
+  let setupId = event.params.preparedSetupId.toHexString();
+  let pluginRepo = event.params.pluginSetupRepo.toHexString();
+  let pluginVersionId = `${pluginRepo}_${event.params.versionTag.release}_${event.params.versionTag.build}`;
+  let installationId = getPluginInstallationId(dao, plugin);
+  if (!installationId) {
+    log.error('Failed to get installationId', [dao, plugin]);
+    return;
+  }
+
+  let preparationId = `${setupId}_${installationId.toHexString()}`;
```

Comment:
Oh boy, it gets confusing with all similar sounding IDs... 😅
No good solution to this problem. 
Should we append `S` and `C` to discriminate `SId`s for the subgraph and `CId`s for the contracts? Not to be confused with IPFS CIDs, of course. 🙃